hail/Makefile

```diff
@@ -267,14 +267,35 @@ resources := $(wildcard python/hailtop/hailctl/dataproc/resources/*)
 $(eval $(call ENV_VAR,cloud_base))
 $(eval $(call ENV_VAR,wheel_cloud_path))
 
+
+
+python/hailtop/pinned-requirements.txt: python/hailtop/requirements.txt
+	HAIL_HAIL_DIR=$(HAIL_HAIL_DIR) \
+	../generate_pip_lockfile.sh python/hailtop/requirements.txt python/hailtop/pinned-requirements.txt
+
+python/pinned-requirements.txt: python/hailtop/requirements.txt python/requirements.txt
```

Comment:
`python/requirements.txt` includes those requirements from `python/hailtop/requirements.txt`, and the dev requirements are constrained by `python/requirements.txt`, so if any of these requirements files changed, it could affect the pinned-requirements.